### PR TITLE
Elvis - check branch, new regexp

### PR DESCRIPTION
**Pull request by bartekgorny**
This PR does two things:
- option to check all files which were modified in a branch (run it before making PR)
- adds another variable name regexp, commented for now, which will be required when we install new version of elvis

### CODE REVIEW

File: elvis.config [hunk 1]
```diff
@@ -8,7 +8,14 @@
...
        rules => [{elvis_style, line_length, #{limit => 100,
                                               skip_comments => false}},
-		 {elvis_style, dont_repeat_yourself, #{min_complexity => 20}}
+                 {elvis_style,
+                  variable_naming_convention,
+                  #{regex => "^([_A-Z][0-9a-zA-Z]*)$" }},
+%%                provision for newer version of elvis_core which doesn't automagically
+%%                strip leading underscore
+%%                https://github.com/inaka/elvis_core/commit/70fd9d01fd076f4640da09fdfd53b310ec59a94c
+%%                  #{regex => "^_{0,2}([A-Z][0-9a-zA-Z]*)$"}},
```
Comment (bartekgorny) [inline elvis.config]: After some hesitation, I think the best way would be to simply state we use Elvis 0.3.2, and when 0.3.3. is out switch to it and modify config file accordingly.

Comment (michalwski) [inline elvis.config]: After another thought, do we know which Elvis version is used by @gadgetci?

Comment (bartekgorny) [inline elvis.config]: Yes We Can

**Comment by michalwski** [comment]
One more question. I observed that ElvisCi is checking only the changes you've made in a PR, not whole changed files. Does the `rock` command work the same way or rather on whole files?

File: elvis.config [hunk 2]
```diff
@@ -8,7 +8,10 @@
        ruleset => erl_files,
        rules => [{elvis_style, line_length, #{limit => 100,
                                               skip_comments => false}},
-		 {elvis_style, dont_repeat_yourself, #{min_complexity => 20}}
+                 {elvis_style,
+                  variable_naming_convention,
+                  #{regex => "^_{0,2}([A-Z][0-9a-zA-Z]*)$"}},
```
Comment (michalwski) [inline elvis.config]: Does it mean that we allow **usage** of vars starting with `_`?

**Comment by bartekgorny** [comment]
`rock` checks the whole files in which you've made any changes. By doing this it forces you to fix coding style in old code too. Hopefully, over time, we'll fix coding style in the whole project.

File: elvis.config (hunk 2 above)
Comment (bartekgorny) [inline elvis.config]: You mean single underscore, or double? Elvis by default allows single underscore, we allow double underscore for reasons already mentioned.

**Pull request merged**